ledger-core/virtual/integration/delegated_finished_test.go

```diff
@@ -73,234 +63,3 @@ func mockExecutor(t *testing.T, server *utils.Server, callMethod callMethodFunc,
-	defer server.Stop()
-
-	runnerMock := logicless.NewServiceMock(ctx, mc, nil)
-	server.ReplaceRunner(runnerMock)
-	server.Init(ctx)
-	server.IncrementPulseAndWaitIdle(ctx)
-
-	gotDelegatedRequestFinished := make(chan *payload.VDelegatedRequestFinished, 1)
-
-	{
-		typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
-		typedChecker.SetDefaultResend(true)
-		typedChecker.VDelegatedRequestFinished.Set(func(finished *payload.VDelegatedRequestFinished) bool {
-			gotDelegatedRequestFinished <- finished
-			return false
-		})
-		typedChecker.VCallResult.Set(func(result *payload.VCallResult) bool { return false })
-	}
-
-	var (
-		class        = testwallet.GetClass()
-		objectLocal  = server.RandomLocalWithPulse()
-		objectGlobal = reference.NewSelf(objectLocal)
-		newState     = makeRawWalletState(initialBalance + 100)
-	)
-
-	Method_PrepareObject(ctx, server, payload.Ready, objectGlobal)
-
-	{
-		pl := payload.VCallRequest{
-			CallType:            payload.CTMethod,
-			CallFlags:           payload.BuildCallFlags(contract.CallTolerable, contract.CallDirty),
-			Caller:              server.GlobalCaller(),
-			Callee:              objectGlobal,
-			CallSiteDeclaration: class,
-			CallSiteMethod:      "AddAmount",
-			CallOutgoing:        server.BuildRandomOutgoingWithPulse(),
-			Arguments:           insolar.MustSerialize([]interface{}{}),
-		}
-
-		newObjectDescriptor := descriptor.NewObject(reference.Global{}, objectLocal, class, []byte(""), reference.Global{})
-
-		result := requestresult.New(makeEmptyResult(), objectGlobal)
-		result.SetAmend(newObjectDescriptor, newState)
-
-		key := pl.CallOutgoing.String()
-		runnerMock.AddExecutionMock(key).
-			AddStart(func(execution execution.Context) {
-				server.IncrementPulse(ctx)
-			}, &execution.Update{
-				Type:   execution.Done,
-				Result: result,
-			})
-		runnerMock.AddExecutionClassify(key, contract.MethodIsolation{
-			Interference: contract.CallTolerable,
-			State:        contract.CallDirty,
-		}, nil)
-
-		beforeCount := server.PublisherMock.GetCount()
-		server.SendPayload(ctx, &pl)
-		if !server.PublisherMock.WaitCount(beforeCount+5, 10*time.Second) {
-			require.Fail(t, "timeout waiting for message")
-		}
-	}
-
-	server.WaitActiveThenIdleConveyor()
-
-	select {
-	case delegateFinishedMsg := <-gotDelegatedRequestFinished:
-		callFlags := payload.BuildCallFlags(contract.CallTolerable, contract.CallDirty)
-
-		require.Equal(t, objectGlobal, delegateFinishedMsg.Callee)
-		require.Equal(t, payload.CTMethod, delegateFinishedMsg.CallType)
-		require.Equal(t, callFlags, delegateFinishedMsg.CallFlags)
-
-		require.Equal(t, newState, delegateFinishedMsg.LatestState.State)
-	default:
-		require.Fail(t, "unexpected situation")
-	}
-}
-
-// In this case VDelegatedRequestFinished must not contain new state since it was not changed
-func TestVirtual_SendDelegatedFinished_IfPulseChanged_Without_SideEffect(t *testing.T) {
-	t.Log("C4990")
```

Comment:
There is a better test with the same scenario: `TestVirtual_Method_PulseChanged`, caseID = "C5211"